rbd: Enable featuregates for volume expansion recovery

humblec opened the pull request:
The enabled feature gate helps us to recover from volume expansion
failure.

Additional Reference#  

Reference below URL for this feature gate details:
https://github.com/kubernetes-csi/external-resizer/

Additional Note for reviewer:

No E2E as its a generic one and not specific to the Ceph CSI driver as like `AnnotateFSResize`. 

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
deploy: Enable featuregate for volume expansion recovery

The enabled feature gates helps us to recover from volume expansion
failure.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

pkalever commented on deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml:
```diff
@@ -104,6 +104,8 @@ spec:
             - "--retry-interval-start=500ms"
             - "--handle-volume-inuse-error=false"
+            #  set it to true to try volume expansion recovery
+            - "--feature-gates=RecoverVolumeExpansionFailure=true,AnnotateFsResize=true"
```
use label `deploy` in the commit-msg instead of `rbd`

humblec replied:
@pkalever this is a draft PR :) , I will be rearranging the commits

pkalever replied:
Yes, I'm aware that it is a draft. Just noticed it, so ... :-)

humblec replied:
sure..thanks..

humblec committed:
helm: enable RecoverVolumeExpansionFailure feature gate

This commit enable the mentioned feature gate which helps to
recover from volume expansion failures.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

Madhu-1 reviewed (commented):
changes LGTM but @humblec What are the cases the volume expansion can fail in cephcsi?

humblec commented:
> changes LGTM but @humblec What are the cases the volume expansion can fail in cephcsi?

it could be anything, more or less, this is a fallback mechanism for the user to come back from an error scenario.

humblec commented:
@Mergifyio rebase

Madhu-1 commented on scripts/minikube.sh:
```diff
@@ -226,7 +226,7 @@ up)
     KUBE_MAJOR=$(kube_version 1)
     KUBE_MINOR=$(kube_version 2)
-    if [ "${KUBE_MAJOR}" -eq 1 ] && [ "${KUBE_MINOR}" -ge 22 ];then
+    if [ "${KUBE_MAJOR}" -eq 1 ] && [ "${KUBE_MINOR}" -eq 22 ];then
```
why this change is required?

Madhu-1 commented on scripts/minikube.sh:
```diff
@@ -235,6 +235,10 @@ up)
         K8S_FEATURE_GATES="${K8S_FEATURE_GATES},PodSecurity=false"
     fi
+    if [ "${KUBE_MAJOR}" -eq 1 ] && [ "${KUBE_MINOR}" -ge 23 ];then
+        # if kubernetes version is greater than 1.22 enable RWOP feature gate
```
comment is not matching the featureGate which is getting enabled.

Madhu-1 commented on deploy/cephfs/kubernetes/csi-cephfsplugin-provisioner.yaml:
```diff
@@ -68,6 +68,8 @@ spec:
             - "--leader-election"
             - "--retry-interval-start=500ms"
             - "--handle-volume-inuse-error=false"
+            #  set it to true to try volume expansion recovery
```
remove the comment. this is not required, flag is self explanatory

humblec committed:
deploy: enable RecoverVolumeExpansionFailure featuregate

this commit enable RecoverVolumeExpansionFailure featuregate in
minikube deployment.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
@Madhu-1 comments addressed, ptal. thanks.

humblec commented:
@Madhu-1 ptal. comments are addressed

humblec commented:
@Mergifyio refresh
